crates/biome_analyze/src/rule.rs

```diff
@@ -42,6 +46,47 @@ pub enum FixKind {
     Unsafe,
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub enum Source {
+    Clippy(&'static str),
+    Eslint(&'static str),
+    EslintJest(&'static str),
+    EslintJsxA11y(&'static str),
+    EslintReact(&'static str),
+    EslintReactHooks(&'static str),
+    EslintTypeScript(&'static str),
+    EslintUnicorn(&'static str),
+    EslintMysticatea(&'static str),
+}
+
+impl std::fmt::Display for Source {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let (source_url, source_rule_name) = match self {
+            Self::Clippy(rule_name) => (format!("https://rust-lang.github.io/rust-clippy/master/#/{rule_name}"), rule_name),
+            Self::Eslint(rule_name) => (format!("https://eslint.org/docs/latest/rules/{rule_name}"), rule_name),
+            Self::EslintJest(rule_name) => (format!("https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/{rule_name}.md"), rule_name),
+            Self::EslintJsxA11y(rule_name) => (format!("https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/{rule_name}.md"), rule_name),
+            Self::EslintReact(rule_name) => (format!("https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/{rule_name}.md"), rule_name),
+            Self::EslintReactHooks(rule_name) => (format!("https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/README.md"), rule_name),
+            Self::EslintTypeScript(rule_name) => (format!("https://typescript-eslint.io/rules/{rule_name}"), rule_name),
+            Self::EslintUnicorn(rule_name) => (format!("https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/{rule_name}"), rule_name),
+            Self::EslintMysticatea(rule_name) => (format!("https://github.com/mysticatea/eslint-plugin/blob/master/docs/rules/{rule_name}"), rule_name),
+        };
```

Comment:
I think we want to make this struct a bit more flexible.

We can have two methods called `Source::as_url`, `Source::as_rule_name` and `Source::as_url_and_rule_name` - internally this method will use `as_url` and `as_rule_name` and return two `&str`. We remove the `Display` implementation from here.

Then, in the `lintdoc`, we will use the method `as_url_and_rule_name` and print the HTML.

Reply:
```rust
impl Source {
    pub fn as_rule_name(&self) -> &'static str {
        match self {
            Self::Clippy(rule_name) => rule_name,
            Self::Eslint(rule_name) => rule_name,
            Self::EslintJest(rule_name) => rule_name,
            Self::EslintJsxA11y(rule_name) => rule_name,
            Self::EslintReact(rule_name) => rule_name,
            Self::EslintReactHooks(rule_name) => rule_name,
            Self::EslintTypeScript(rule_name) => rule_name,
            Self::EslintUnicorn(rule_name) => rule_name,
            Self::EslintMysticatea(rule_name) => rule_name,
        }
    }

    pub fn as_url(&self) -> &'static str {
        match self {
            Self::Clippy(_) =>"https://rust-lang.github.io/rust-clippy/master/index.html/#/",
            Self::Eslint(_) =>"https://eslint.org/docs/latest/rules/",
            Self::EslintJest(_) =>"https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/",
            Self::EslintJsxA11y(_) =>"https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/",
            Self::EslintReact(_) =>"https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/",
            Self::EslintReactHooks(_) =>"https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/README.md",
            Self::EslintTypeScript(_) =>"https://typescript-eslint.io/rules/",
            Self::EslintUnicorn(_) =>"https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/",
            Self::EslintMysticatea(_) =>"https://github.com/mysticatea/eslint-plugin/blob/master/docs/rules/",
        }
    }

    pub fn as_url_and_rule_name(&self) -> (&'static str, &'static str) {
        (self.as_url(), self.as_rule_name())
    }
}

```

I'm trying to do it this way here, but I can't manage the urls that need to end in .md. Is there any other way to suggest or how to resolve it?

Reply:
The method `as_url` should return the URL **of the rule**. Maybe we should call the method `as_rule_url`. 

I don't think we have a case where we need to link the homepage of the source.

Reply:
I made it work using the macro format. I couldn't think of any other way.

I pushed the commit with the changes.